### ISSUE TITLE
Switch to Netplan (netplan) for network configuration on Ubuntu

### ISSUE THREAD

**Issue by lmakarov** [docksal/docksal#412]
Previous issue: #321 

Research using [netplan](https://wiki.ubuntu.com/Netplan) for network management on Ubuntu.

Once it lands in Ubuntu 18.04 LTS, we should stop using the old tools (`ifup`, `ifdown`, `resolvconf`) and use netplan instead.

**Comment by achekulaev** [comment]
I guess we will have to differentiate Ubuntu 18.04 vs older Ubuntu/Debian versions.

**Comment by lmakarov** [comment]
Once Ubuntu 18.04 LTS is out, that would be THE LTS we support (so dropping 16.04 LTS).
Regular users will always be on the latest Ubuntu versions anyway.

**Comment by theinfiltrated** [comment]
What about openvswitch support? I cannot live without it.

**Comment by frederickjh** [comment]
Not sure dropping support for Ubuntu 16.04 LTS before its end of life in April of 2021 in such a wise decision. 
The Ubuntu version can be gotten from `lsb_release -r`. We have already seen that the `docker-ce` package is not yet available for 18.04. So, there could be other reasons that people wait to upgrade.

**Comment by lmakarov** [comment]
@frederickjh if 18.04 works with the old (currently used in fin) network configuration tools/configs, then we don't even have to change anything.

I have not done any testing with 18.04 yet. A comparison like [here](https://github.com/docksal/docksal/issues/321#issuecomment-349081787) would be helpful.

**Comment by frederickjh** [comment]
I too have not done any testing on 18.04. I am holding off as I use the LTS versions. Normally they do not prompt LTS users to update until the **xx.xx.1** release comes out. 18.04.1 will come out some time in June 2018. That and the fact that docker-ce will not be available until around the same time.

**Comment by achekulaev** [comment]
18.04 does not work with the old approach.
`ifdown` freezes waiting for lock `ifdown: waiting for lock on /run/network/ifstate.lo`

I checked if alpine functions work, and they do work. Seems like a good moment to switch over to them.

**Comment by lmakarov** [comment]
Agreed. Closing in favor of #613
